operator/src/main/java/org/bf2/operator/events/ResourceEventSource.java

```diff
@@ -47,11 +47,10 @@ protected void handleEvent(HasMetadata resource, ResourceAction action) {
             if(resource.getMetadata().getOwnerReferences().isEmpty()) {
                 log.warnf("%s %s/%s does not have OwnerReference", resource.getKind(), resource.getMetadata().getNamespace(), resource.getMetadata().getName());
             } else {
-                getEventHandler().handleEvent(new ResourceEvent(action, ResourceID.fromFirstOwnerReference(resource).get()));
+                getEventHandler().handleEvent(new ResourceEvent(action, ResourceID.fromFirstOwnerReference(resource).orElse(null)));
```

Comment:
Try something like this to handle the `Optional`:
```java
ResourceID.fromFirstOwnerReference(resource).ifPresentOrElse(
    ownerId -> getEventHandler().handleEvent(new ResourceEvent(action, ownerId)),
    () -> log.warnf("%s %s/%s does not have OwnerReference", resource.getKind(), resource.getMetadata().getNamespace(), resource.getMetadata().getName()));
```

Reply:
@MikeEdgar This one is throwing an error of incorrect type. `Required type: ResourceID, Provided: void`

Reply:
The code snippet should be the full contents within `if (getEventHandler() != null) { ... }`

Reply:
That worked!